app/models/piece.rb

```diff
@@ -12,9 +14,18 @@ class Piece < ApplicationRecord
   #
   # def move(row,col)
   # end
-  #
-  # def is_obstructed?(dest_row, dest_col)
-  # end
+
+  def obstructed?(dest_row, dest_col)
+    if dest_row == self.x_position
```

Comment:
I'm not sure about dest_row == self.x_position line
If we are moving horizontally then dest_row would be a new desired position right? And it can't be equal to the
self.x_position...

Reply:
The logic I was thinking about this is if a piece is obstructed, then it cannot move, so its x_position will remain the same.

Reply:
I'd think this is to figure out wether we are moving vertically or horizontally. But it will only make sense if for horizontal obstruction you compare y destination ( dest_row in your case )  and current y position. That means that the Y positions stays the same when we move and we are actually moving horizontally. 
The same with moving vertically if X destination ( col ) the same like current X that means we are moving vertically. In chess row is going vertically and column horizontally.

Reply:
Also I think it's a good practice to keep the same names within our code - I guess it doesn't really matter if we call it dest_row or destination_y as long as we  call it the same name everywhere.

Reply:
Yes, you are definitely right, I will change the variables to destination_x and destination_y when I get home.

For the other logic, I am thinking if a piece is obstructed, then it can't move, that's why it's destination_x or destination_y remains the same, because the piece did not move anywhere. I think of it like the obstructing piece is blocking the way of the piece, stopping the piece from moving at all. If the piece can move either horizontally or vertically then it's not blocked, it's moving.